PD-1141: Adds logos to logo component

brookescarlett opened the pull request:
<!--
Thanks for contributing to LeafyGreen!

Before you submit your pull request, please be sure that you've reviewed our contributing guidelines: https://github.com/mongodb/leafygreen-ui/blob/main/DEVELOPER.md

Please fill out the information below to help speed the review along, and hopefully
the merge of your pull request!
-->

## ✍️ Proposed changes

- Adds AtlasLogo/LogoMark, ChartsLogo/LogoMark, CloudManager Logo/LogoMark, Realm Logo/LogoMark, Server LogoMark, Drivers LogoMark, Compass LogoMark
- Updates Logo with `product` and `lockup` props 

🎟 _Jira ticket:_ [PD-1141](https://jira.mongodb.org/browse/PD-1141)

## 🛠 Types of changes

<!--
What types of changes does your code introduce? Put an `x` in the applicable boxes.
-->

- [ ] Tooling (updates to workspace config or internal tooling)
- [ ] Bugfix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [x] Breaking change (fix or feature that would cause existing functionality to not work as expected)

## ✅ Checklist

<!--
Put an `x` in the boxes that apply. You can also fill these out after creating the PR. If you're unsure about any of them, don't hesitate to ask. We're here to help! This is simply a reminder of what we are going to look for before merging your code.
-->

- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] I have added necessary documentation (if appropriate)
- [ ] I have run `yarn changeset` and documented my changes

<!--
The following only apply when building a new component
-->

- [ ] I have added my new package to the global tsconfig
- [ ] I have added my new package to the Table of Contents on the global README

## 💬 Further comments

<!--
If this is a relatively large or complex change, kick off the discussion by explaining why you chose the solution you did and what alternatives you considered, etc...

Consider putting screenshots of your addition / change here if there are visual changes, and a gif if motion is a major component of it.

Alternatively, if this is a very minor, and self-explanatory change, feel free to remove this section.
-->

DesignerDave reviewed (changes requested):
Can we add the stacked prop to storybook?

I noticed we changed variant to darkMode for logo marks but not the main logo. Can we also change that there?

Also changeset 😄

brookescarlett commented:
> Can we add the stacked prop to storybook?
> 
> I noticed we changed variant to darkMode for logo marks but not the main logo. Can we also change that there?
> 
> Also changeset 😄

- Stacked prop already here: https://github.com/mongodb/leafygreen-ui/pull/833/files#diff-da5fae2adce5b44126f4c38f45d9065e4d16add014182ad204bab454cc5cbda5R95
- Removed the variant from the readme

DesignerDave reviewed (approved):
Totally missed the knob 🤦 

LGTM!
